### PR TITLE
Remove unused headers

### PR DESCRIPTION
**Pull request by kinkard**
# Issue

This PR removes unused C++ headers. There is no direct value from that aside small build time speed up, which is always nice

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by kevinkreiser** [comment]
very nice many thanks for this! we always wanted to do the include what you use cleanup, i think its mentioned somewhere in some issue. would be great to have a linter rule for this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Unreleased
 * **Removed**
+   * REMOVED: unused headers [#4829](https://github.com/valhalla/valhalla/pull/4829)
```
Comment (kevinkreiser) [inline CHANGELOG.md:3]: we do the changelog entries such that new ones are at the bottom of the current unreleased section

File: src/baldr/streetname_us.cc [hunk 2]
```diff
@@ -1,7 +1,5 @@
-#include <iostream>
-
-#include "baldr/streetname.h"
 #include "baldr/streetname_us.h"
+#include "baldr/streetname.h"
```
Comment (kevinkreiser) [inline src/baldr/streetname_us.cc:2]: ill be shocked if the linter doesnt complain about reordering these (i think its alphabetically sorts them)

Comment (kevinkreiser) [reply src/baldr/streetname_us.cc:2]: strange, i guess it doesnt care about sorting then!

Comment (kinkard) [reply src/baldr/streetname_us.cc:2]: I believe `clang-format` has a specific rule for the header that has the same name as the current source file.

**Comment by kinkard** [comment]
I used really opiniated linter for this cleanup - https://gist.github.com/kinkard/f0d41b4e5f1f29347b8bdb4cb723defa

**Pull request merged**